pkg/volume/rbd/rbd.go

```diff
@@ -29,17 +29,20 @@ import (
 )
 
 // This is the primary entrypoint for volume plugins.
-func ProbeVolumePlugins() []volume.VolumePlugin {
-	return []volume.VolumePlugin{&rbdPlugin{nil, exec.New()}}
+func ProbeVolumePlugins(volumeConfig volume.VolumeConfig) []volume.VolumePlugin {
+	return []volume.VolumePlugin{&rbdPlugin{nil, exec.New(), newRecycler, volumeConfig}}
 }
 
 type rbdPlugin struct {
-	host volume.VolumeHost
-	exe  exec.Interface
+	host            volume.VolumeHost
+	exe             exec.Interface
+	newRecyclerFunc func(*volume.Spec, volume.VolumeHost, volume.VolumeConfig) (volume.Recycler, error)
```

Comment:
The idea is to allow a test to supply the factory function.  This used to be in more plugins w/ builder and cleaner but they've been refactored out.

I agree that this indirection (complexity) can be removed if it is not used anywhere.